pyanaconda/modules/payloads/payload/payload_base.py

```diff
@@ -159,6 +120,19 @@ def is_network_required(self):
 
         return False
 
+    def calculate_required_space(self):
+        """Calculate space required for the installation.
+
+        :return: required size in bytes
+        :rtype: int
+        """
+        total = 0
+
+        for source in self.sources:
+            total += source.required_space
+
+        return total
```

Comment:
Are you aware that this will never work for DNF payload and probably even Flatpak and OSTree (not that sure there)?

DNF have to read package headers to get info about the size, that is done by dnf library not by us.

Reply:
Yes, but I still think that this is the most universal default. This method is very similar to `is_network_required` above.

Reply:
Good point, we don't have better default.